modules/tinymce/src/plugins/paste/test/ts/browser/SmartPasteTest.ts

```diff
@@ -35,19 +36,78 @@ UnitTest.asynctest('browser.tinymce.plugins.paste.SmartPasteTest', (success, fai
     LegacyUnit.equal(SmartPaste.isAbsoluteUrl(''), false);
   });
 
-  suite.test('TestCase-TBA: Paste: isImageUrl', function () {
-    LegacyUnit.equal(SmartPaste.isImageUrl('http://www.site.com'), false);
-    LegacyUnit.equal(SmartPaste.isImageUrl('https://www.site.com'), false);
-    LegacyUnit.equal(SmartPaste.isImageUrl('http://www.site.com/dir-name/file.jpeg'), true);
-    LegacyUnit.equal(SmartPaste.isImageUrl('http://www.site.com/dir-name/file.jpg'), true);
-    LegacyUnit.equal(SmartPaste.isImageUrl('http://www.site.com/dir-name/file.png'), true);
-    LegacyUnit.equal(SmartPaste.isImageUrl('http://www.site.com/dir-name/file.gif'), true);
-    LegacyUnit.equal(SmartPaste.isImageUrl('https://www.site.com/dir-name/file.gif'), true);
-    LegacyUnit.equal(SmartPaste.isImageUrl('https://www.site.com/~dir-name/file.gif'), true);
-    LegacyUnit.equal(SmartPaste.isImageUrl('https://www.site.com/dir-name/file.gif?query=%42'), false);
-    LegacyUnit.equal(SmartPaste.isImageUrl('https://www.site.com/dir-name/file.html?query=%42'), false);
-    LegacyUnit.equal(SmartPaste.isImageUrl('file.gif'), false);
-    LegacyUnit.equal(SmartPaste.isImageUrl(''), false);
+  suite.test('TestCase-TBA: Paste: isImageUrl', function (editor) {
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'http://www.site.com'), false);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'https://www.site.com'), false);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'http://www.site.com/dir-name/file.jpeg'), true);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'http://www.site.com/dir-name/file.jpg'), true);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'http://www.site.com/dir-name/file.png'), true);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'http://www.site.com/dir-name/file.gif'), true);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'https://www.site.com/dir-name/file.gif'), true);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'https://www.site.com/~dir-name/file.gif'), true);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'https://www.site.com/dir-name/file.gif?query=%42'), false);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'https://www.site.com/dir-name/file.html?query=%42'), false);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, 'file.gif'), false);
+    LegacyUnit.equal(SmartPaste.isImageUrl(editor, ''), false);
+  });
+
+  suite.test('TINY-6306: New image_file_types defaults', (editor) => {
+    Arr.map([
+      'jpeg',
+      'jpg',
+      'jpe',
+      'jfi',
+      'jfif',
+      'png',
+      'gif',
+      'bmp',
+      'webp'
+    ], (image_file_type) => LegacyUnit.equal(
```

Comment:
Based on the potential `.` escaping thing I commented on earlier, can we figure out the correct behaviour for a url like `https://www.site.com/filejpeg` and then add an assertion for that here?